### PR TITLE
qa/tasks/ceph_salt: Add Containers as parameters

### PR DESCRIPTION
**Pull request by gekios**
Add registries and containers via yaml files in order to be able to have
different ones for each case via teuthology overrides

Signed-off-by: Georgios Kyratsas <gkyratsas@suse.com>

Requires https://gitlab.suse.de/ceph/storage-ci/-/merge_requests/147 to be merged

### CODE REVIEW

**Comment by kshtsk** [comment]
retest this please

File: qa/tasks/ceph_salt.py [hunk 1]
```diff
@@ -61,15 +61,44 @@ class CephSalt(Task):
...
+    There is also the possibility of using overrides in order to set the
+    container registry and image like:
+
+        overrides:
+            ceph_salt:
+                containers:
+                    registry:
+                        name: 'registry.mirror.example.com'
+                        location: 'registry.mirror.example.com:5000'
+                    image: 'quay.io/ceph-ci/ceph'
```
Comment (kshtsk) [inline qa/tasks/ceph_salt.py:90]: This looks like a copy-paste from the cephadm task?

File: qa/tasks/ceph_salt.py [hunk 2]
```diff
@@ -61,15 +61,44 @@ class CephSalt(Task):
...
+            registry:
+                name: registry name
+                location: <registry url/IP:port>
+                insecure: True/False    (Is registry insecure, True by default)
+            image:  <container image path>
 
+
+    For example:
+
+    containers:
```
Comment (kshtsk) [inline qa/tasks/ceph_salt.py:74]: For the first line there should extra indent.

File: qa/tasks/ceph_salt.py [hunk 3]
```diff
@@ -61,15 +61,44 @@ class CephSalt(Task):
                         using zypper.
         branch:         Ceph-salt branch in case repo is provided. If no branch
                         is provided master is used by default.
+        containers:
+            registry:
+                name: registry name
+                location: <registry url/IP:port>
```
Comment (kshtsk) [inline qa/tasks/ceph_salt.py:67]: IP is already a part of url

File: qa/tasks/ceph_salt.py (hunk 1 above)
Comment (gekios) [reply qa/tasks/ceph_salt.py:90]: yep since there is a plan to move it upstream I just used this as generic example not to have suse only

**Comment by kshtsk** [comment]
@susebot run leap makecheck

**Comment by kshtsk** [comment]
@susebot run leap mkck

File: qa/tasks/ceph_salt.py [hunk 4]
```diff
@@ -61,15 +61,44 @@ class CephSalt(Task):
                         using zypper.
         branch:         Ceph-salt branch in case repo is provided. If no branch
                         is provided master is used by default.
+        containers:
+            registry:
+                name: registry name
+                location: <registry url:port>
```
Comment (kshtsk) [inline qa/tasks/ceph_salt.py:67]: port number is already a part of url, you probably meant: `hostname:port`?

File: qa/tasks/ceph_salt.py (hunk 1 above)
Comment (kshtsk) [reply qa/tasks/ceph_salt.py:90]: the thing is that `image` value in cephadm is used in a different way, the append :sha1 value to it, not sure if we have the same schema

Comment (kshtsk) [reply qa/tasks/ceph_salt.py:90]: So, I think it can misguide a reader

**Comment by kshtsk** [comment]
@susebot run sles mkck

**Comment by kshtsk** [comment]
@susebot run sles mkck

File: qa/tasks/ceph_salt.py (hunk 1 above)
Comment (gekios) [reply qa/tasks/ceph_salt.py:90]: To my pov image is an image, the fact that there can be sha1 or not in the name/path is irrelevant so I don't see anything wrong here.

File: qa/tasks/ceph_salt.py [hunk 5]
```diff
@@ -61,15 +61,44 @@ class CephSalt(Task):
                         using zypper.
         branch:         Ceph-salt branch in case repo is provided. If no branch
                         is provided master is used by default.
+        containers:
+            registry:
+                name: registry name
+                location: <registry url>
+                insecure: True/False    (Is registry insecure, True by default)
+            image:  <container image path>
```
Comment (toabctl) [inline qa/tasks/ceph_salt.py:69]: maybe call this `image_ceph`? I think there are more images (eg. one for monitoring)

Comment (gekios) [reply qa/tasks/ceph_salt.py]: done, I renamed it to `ceph_image`

**Comment by gekios** [comment]
https://gitlab.suse.de/ceph/storage-ci/-/merge_requests/147 was merged so to my POV we can merge this PR as well unless there is any objection :)

**Pull request merged**